routes/api/promotions.js

```diff
@@ -15,6 +16,23 @@ const promotionsRouter = Router();
  * default value is ['static', 'popup'] (any promotion)
  */
 
+promotionsRouter.get('/public-ad', async (req, res) => {
+  if (await Promotion.findOne()) {
+    const cont = 1;
+    const singleAd = await Promotion.findOne({ order: random});
+    let valorActual = singleAd.dataValues.requestedQuantities;
+    let valorNuevo = valorActual + cont;
+    singleAd.update({ requestedQuantities: valorNuevo });
+    return res.status(200).json({ success: true, data: singleAd });
```

Comment:
1. Hace falta el await
2. Agreguemos un try y un catch por si acaso
3. Usa DTOs de Daikone
```suggestion
    try {
        await singleAd.update({ requestedQuantities: requestedQuantity });
        return res.status(200).json({ success: true, data: singleAd }); // USAR DTOs
    } catch {
        return res.status(500).json({ success: false, message: "Error desconocido al actualizar la cantidad de renderizaciones del anuncio" })
    }
```

Otra cosa...
4. ¿No deberia ser "item"?, ¿como esta en la doc?
```suggestion
    try {
        await singleAd.update({ requestedQuantities: requestedQuantity });
        return res.status(200).json({ success: true, item: singleAd }); // USAR DTOs
    } catch {
        return res.status(500).json({ success: false, message: "Error desconocido al actualizar la cantidad de renderizaciones del anuncio" })
    }
```

Reply:
?